### PR TITLE
Custom fetch all handler for vertica to not miss errors

### LINKED ISSUE #32993: Error not detected in multi-statement vertica query
### Apache Airflow version

2.6.3

### What happened

Hello, 

There is a problem with multi-statement query and vertica, error will be detected only if it happens on the first statement of the sql.

for example if I run the following sql with default SQLExecuteQueryOperator options:
INSERT INTO MyTable (Key, Label) values (1, 'test 1');
INSERT INTO MyTable (Key, Label) values (1, 'test 2');
INSERT INTO MyTable (Key, Label) values (3, 'test 3');

the first insert will be commited, the nexts won't and no errors will be returned.
the same sql runed on mysql will return an error and no row will be inserted.

It seems to be linked to the way the vertica python client works (an issue has been opened on their git 4 years ago, [Duplicate key values error is not thrown as exception and is getting ignored](https://github.com/vertica/vertica-python/issues/255)) but since a workaround was provided I don't think it will be fixed in a near future. 

For the moment, to workaroud I use the split statement option with disabling auto-commit but I think it's dangerous to let this behaviour as is.

### What you think should happen instead

_No response_

### How to reproduce

create a table MyTable with two columns Key and Lbl, declare Key as primary key.
Run the following query with SQLExecuteQueryOperator 
INSERT INTO MyTable (Key, Label) values (1, 'test 1');
INSERT INTO MyTable (Key, Label) values (1, 'test 2');
INSERT INTO MyTable (Key, Label) values (3, 'test 3');

### Operating System

Debian GNU/Linux 11 (bullseye)

### Versions of Apache Airflow Providers

_No response_

### Deployment

Official Apache Airflow Helm Chart

### Deployment details

_No response_

### Anything else

_No response_

### Are you willing to submit PR?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by darkag**
Change the default fetch all handler for vertica provider in order to catch all errors in multi-statement query ([issue 32993](https://github.com/apache/airflow/issues/32993))
Closes: https://github.com/apache/airflow/issues/32993

### CODE REVIEW

File: airflow/providers/vertica/hooks/vertica.py [hunk 1]
```diff
@@ -17,9 +17,24 @@
...
+
 from vertica_python import connect
 
-from airflow.providers.common.sql.hooks.sql import DbApiHook
+from airflow.providers.common.sql.hooks.sql import DbApiHook, fetch_all_handler
+
+
+def vertica_fetch_all_handler(cursor) -> list[tuple] | None:
+    """Replace the default DbApiHook fetch_all_handler ."""
+    to_return = fetch_all_handler(cursor)
```
Comment (Lee-W) [inline airflow/providers/vertica/hooks/vertica.py]: Could we rename `to_return`? Does `rows` make sense as the name of this value? or `result` might be a bit better than `to_return`

File: airflow/providers/vertica/hooks/vertica.py [hunk 2]
```diff
@@ -17,9 +17,24 @@
...
+    # loop on all statement result sets to get errors
+    if cursor.description is not None:
+        while cursor.nextset():
+            if cursor.description is not None:
+                row = cursor.fetchone()
+                while row:
+                    row = cursor.fetchone()
+    return to_return
```
Comment (Lee-W) [inline airflow/providers/vertica/hooks/vertica.py]: Will this value be changed during the execution of lines 31-36? It seems this variable is not touched after it is assigned. But I guess this is something related to cursor?

Comment (darkag) [reply airflow/providers/vertica/hooks/vertica.py]: Returned value will not change between line 31 to 37, all this code is here to make vertica client throws error. 
If you run the following sql:
INSERT INTO MyTable (Key, Label) values (1, 'test 1');
INSERT INTO MyTable (Key, Label) values (1, 'test 2');
INSERT INTO MyTable (Key, Label) values (3, 'test 3');

each insert has its own result set and if you don't try to fetch data of thoses result sets you won't detect error on the second insert.

File: airflow/providers/vertica/hooks/vertica.py (hunk 1 above)
Comment (darkag) [reply airflow/providers/vertica/hooks/vertica.py]: ok I'll rename it as result

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Interesting ... LGTM.

File: airflow/providers/vertica/hooks/vertica.py (hunk 2 above)
Comment (potiuk) [reply airflow/providers/vertica/hooks/vertica.py]: Very nasty behaviour of Vertica

**Review by potiuk — CHANGES_REQUESTED** [review]
One change request though: : could you please add some comments (docstring) in VerticaHook explaining why we have this and explaining why we have this complexity in the Vertica Hook. It should contain link to the issue that you have in the commit message as well as some explanation for those who would like to use their own handler for Vertica Hook.

I think without it it will be difficult for future maintainers to figure out why we are doing it and for the users to properly write their own handlers if they choose to do so.

**Comment by darkag** [comment]
Added docstring (hope this is understandable enough)

File: airflow/providers/vertica/hooks/vertica.py (hunk 2 above)
Comment (darkag) [reply airflow/providers/vertica/hooks/vertica.py]: Yes, fortunately that as beginning with airflow I tested that it worked as I thincked, but if I had the habit with other database before I may have ended with a really bad surprise.

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Thanks! LGTM!

**Pull request merged**